lib/commands/recordscreen.js

```diff
@@ -0,0 +1,277 @@
+ * @property {?string} user - The name of the user for the remote authentication. Only works if `remotePath` is provided.
+ * @property {?string} pass - The password for the remote authentication. Only works if `remotePath` is provided.
+ * @property {?string} method - The http multipart upload method name. The 'PUT' one is used by default.
+ *                              Only works if `remotePath` is provided.
+ * @property {?string} videoType - The format of the screen capture to be recorded.
+ *                                 Available formats: "h264", "mp4" or "fmp4". Default is "mp4".
+ *                                 Only works for Simulator.
+ * @property {?string} videoQuality - The video encoding quality (low, medium, high, photo - defaults to high).
+ *                                    Only works for real devices.
+ * @property {?boolean} forceRestart - Whether to try to catch and upload/return the currently running screen recording
+ *                                     (`false`, the default setting) or ignore the result of it and start a new recording
+ *                                     immediately.
+ * @property {?string|number} timeLimit - The maximum recording time, in seconds.
+ *                                        The default value is 180, the maximum value is 600 (10 minutes).
+ */
+
+/**
+ * Record the display of devices running iOS Simulator since Xcode 8.3 or real devices since iOS 8
+ * (xrecord utility is required: https://github.com/WPO-Foundation/xrecord).
+ * It records screen activity to an MPEG-4 file. Audio is not recorded with the video file.
+ * If screen recording has been already started then the command will stop it forcefully and start a new one.
+ * The previously recorded video file will be deleted.
+ *
+ * @param {?StartRecordingOptions} options - The available options.
+ * @returns {string} Base64-encoded content of the recorded media file if
+ *                   any screen recording is currently running or an empty string.
+ * @throws {Error} If screen recording has failed to start.
+ */
+commands.startRecordingScreen = async function (options = {}) {
+  const {videoType, timeLimit, videoQuality, forceRestart} = options;
+
+  let result = '';
+  if (!forceRestart) {
+    result = await this.stopRecordingScreen(options);
+  }
+  const pgrepPattern = this.isRealDevice() ? REAL_DEVICE_PGREP_PATTERN : SIMULATOR_PGREP_PATTERN;
+  const pid = await getPidUsingPattern(pgrepPattern(this.opts.device.udid));
+  if (!_.isEmpty(pid)) {
+    try {
+      await exec('kill', [pid]);
+    } catch (err) {
+      log.errorAndThrow(`Unable to stop screen recording process: ${err.message}`);
+    }
+  }
+  if (!_.isEmpty(this._recentScreenRecordingPath)) {
+    await fs.rimraf(this._recentScreenRecordingPath);
+    this._recentScreenRecordingPath = null;
+  }
+
+  const localPath = temp.path({prefix: 'appium', suffix: DEFAULT_EXT});
```

Comment:
`appium-support` supports this: https://github.com/appium/appium-support/blob/master/lib/tempdir.js

Reply:
shall I use tempdir.path instead? I find it a bit redundant, since I don't need a whole directory to be created, but rather just a single file

Reply:
I would rather tall temporary file management be done in a single place, yes.